src/backend/executor/nodeShareInputScan.c

```diff
@@ -997,15 +993,24 @@ shareinput_writer_notifyready(shareinput_Xslice_reference *ref)
  *
  *  This is a non-blocking operation.
  */
-static void
-shareinput_reader_notifydone(shareinput_Xslice_reference *ref, int nconsumers)
+void
+shareinput_reader_notifydone(ShareInputScanState *state)
```

Comment:
how about to reduce diff below by
```suggestion
shareinput_reader_notifydone(ShareInputScanState *node)
```